src/Features/Core/Portable/ConvertTupleToStruct/AbstractConvertTupleToStructCodeRefactoringProvider.cs

```diff
@@ -913,30 +916,28 @@ private SyntaxNodeOrToken ConvertArgumentOrToken(SyntaxGenerator generator, Nami
             var parameters = fields.SelectAsArray(field =>
             {
                 var parameter = CodeGenerationSymbolFactory.CreateParameterSymbol(
-                    field.Type, GetConstructorParameterName(parameterNamingRule, field.Name));
+                    field.Type, GetConstructorParameterName(isRecord, parameterNamingRule, field.Name));
 
                 parameterToPropMap[parameter.Name] = field;
 
                 return parameter;
             });
 
-            // If we're making a record-struct and the parameters and properties all have the same name, then we can
-            // make a primary constructor here.
-            var isPrimaryConstructor = isRecord && parameterToPropMap.All(kvp => kvp.Key == kvp.Value.Name);
-
             var assignmentStatements = generator.CreateAssignmentStatements(
                 semanticModel, parameters, parameterToPropMap, ImmutableDictionary<string, string>.Empty,
                 addNullChecks: false, preferThrowExpression: false);
 
             var constructor = CodeGenerationSymbolFactory.CreateConstructorSymbol(
                 attributes: default, Accessibility.Public, modifiers: default,
-                className, parameters, assignmentStatements, isPrimaryConstructor: isPrimaryConstructor);
+                className, parameters, assignmentStatements, isPrimaryConstructor: isRecord);
 
-            return (constructor, isPrimaryConstructor);
+            return constructor;
         }
 
-        private static string GetConstructorParameterName(NamingRule parameterNamingRule, string name)
-            => parameterNamingRule.NamingStyle.MakeCompliant(name).First();
+        private static string GetConstructorParameterName(bool isRecord, NamingRule parameterNamingRule, string name)
+            => isRecord
+            ? name
+            : parameterNamingRule.NamingStyle.MakeCompliant(name).First();
```

Comment:
```suggestion
            => isRecord ? name : parameterNamingRule.NamingStyle.MakeCompliant(name).First();
```

Reply:
also, consider just inlining this method entirely, i don't think it serves a ton of purpose.